<launch>
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="image_rect" />
    <remap from="camera_info" to="camera_info" />

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="raw" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <remap from="camera_info" to="rgb/camera_info"/>
    <remap from="cloud_rect" to="depth_filtered/points"/>
    <remap from="tag_detections_pose" to="/sensors/apriltag/detections_pose"/>
    <remap from="tag_detections" to="/sensors/apriltag/detections"/>
    <remap from="tag_detections_image" to="/sensors/apriltag/detections_image"/>
    <remap from="enable" to="/sensors/apriltag/enable"/>

    <param name="queue_size" type="int" value="5"/>
    <param name="start_enabled" type="bool" value="false" />
    <param name="decimate_rate" type="int" value="6" />
    <param name="publish_plane_cloud" type="bool" value="false" />
    <param name="plane_angle_threshold" type="double" value="0.174533" />
    <param name="plane_model_distance_threshold" type="double" value="0.05" />
    <param name="tag_family" type="str" value="36h11"/>
    <param name="projected_optics" type="bool" value="true"/>
    <param name="output_frame_id" type="str" value="base_footprint"/>

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.163513},
      {id: 1, size: 0.163513, frame_id: a_frame},
      {id: 2, size: 0.163513, frame_id: tag_2},
      {id: 3, size: 0.163513},
      {id: 4, size: 0.163513},
      {id: 5, size: 0.163513}]
    </rosparam>
  </node>
</launch>
